src/popup.js

```diff
@@ -2,129 +2,218 @@
 
 import { Status } from "./status.js";
 
+const CONNECTION_ERROR =
+    "Could not establish connection. Receiving end does not exist.";
+
 const ids = {
     status: "status",
-    restore: "restore",
     reloadPage: "reload-page",
+    reloadMessage: "reload-message",
     showChanges: "show-changes",
     showChangesCheckbox: "show-changes-checkbox",
     showHighlights: "show-highlights",
-    showHighlightsCheckbox: "show-highlights-checkbox"
+    showHighlightsCheckbox: "show-highlights-checkbox",
+    turnOnForHost: "turn-on-for-host",
+    turnOnForHostCheckbox: "turn-on-for-host-checkbox"
 };
 
-// Controls the DGtW content script on the current page
-function sendMessageToContentScript(type, callback) {
-    // If this is a test, the popup is opened in a tab. This steals focus from
-    // the tab in which the content script is running.
+function callOnTargetTab(callback) {
     const queryInfo = { active: !isTest, currentWindow: true };
-
     chrome.tabs.query(queryInfo, function(tabs) {
         // Normally, only one tab matches the query.
         // When testing, a blank tab might match the query. Using `length - 1`
         // picks the correct target, the most recently opened background tab.
-        const targetTab = tabs[tabs.length - 1].id;
-        chrome.tabs.sendMessage(targetTab, { type: type }, callback);
+        const tab = tabs[tabs.length - 1];
+        callback(tab);
     });
 }
 
-function showElement(id) {
-    document.getElementById(id).classList.remove("hide");
-    document.getElementById(id).classList.add("show");
+// Controls the DGtW content script on the current page
+function sendMessageToContentScript(type, callback) {
+    // If this is a test, the popup is opened in a tab. This steals focus from
+    // the tab in which the content script is running.
+    callOnTargetTab(function(tab) {
+        chrome.tabs.sendMessage(tab.id, { type: type }, callback);
+    });
 }
 
-function hideElement(id) {
-    document.getElementById(id).classList.remove("show");
-    document.getElementById(id).classList.add("hide");
+function setVisibility(id, visibility) {
+    if (visibility) {
```

Comment:
I like the article! The point of this function is that something will be set to visible or not visible. I wouldn't feel good about `isVisible` because of the implication that it's the current state of the element. I've chosen `shouldBeVisible`, since "is" is a conjugation of "to be". It also adequately implies that setting it to `false` hides the element.